src/main/java/mate/academy/service/impl/BookingServiceImpl.java

```diff
@@ -0,0 +1,207 @@
+import mate.academy.model.User;
+import mate.academy.repository.PaymentRepository;
+import mate.academy.repository.UserRepository;
+import mate.academy.repository.booking.BookingRepository;
+import mate.academy.service.BookingService;
+import mate.academy.service.NotificationService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class BookingServiceImpl implements BookingService {
+    private final BookingMapper bookingMapper;
+    private final BookingRepository bookingRepository;
+    private final NotificationService notificationService;
+    private final PaymentRepository paymentRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    public BookingDto getBookingById(Long id) {
+        return bookingMapper.toDto(bookingRepository.getBookingById(id)
+                .orElseThrow(() -> new
+                        EntityNotFoundException("Can't find booking with id " + id)));
+    }
+
+    @Override
+    public List<BookingDto> getBookingsByUserId(Long userId, Pageable pageable) {
+        return bookingRepository.findAllByUserId(userId, pageable)
+                .stream()
+                .map(bookingMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<BookingDto> getAllBookingsWithoutUserId(Pageable pageable) {
+        Page<Booking> allOrders = bookingRepository.findAll(pageable);
+        return bookingMapper.toBookingDto(allOrders);
+    }
+
+    @Override
+    public BookingDto createBooking(CreateBookingRequestDto request, Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(""
+                        + "User not found with id: " + userId));
+
+        long pendingPayments = paymentRepository.countPendingPaymentsByUserId(userId);
+        if (pendingPayments > 0) {
+            throw new IllegalStateException("You have pending payments. "
+                    + "Please pay them before booking.");
+        }
+
+        for (LocalDate date = request.getCheckInDate();
+                !date.isAfter(request.getCheckOutDate());
+                date = date.plusDays(1)) {
+            boolean isBooked = bookingRepository
+                    .existsByAccommodationIdAndCheckInDateBeforeAndCheckOutDateAfter(
+                            request.getAccommodationId(), date, date);
+            if (isBooked) {
+                throw new BookingAlreadyExistsException(""
+                        + "Accommodation is already booked during the selected period");
+            }
+        }
+
+        Booking booking = bookingMapper.toEntity(request);
+        booking.setUser(user);
+
+        Booking savedBooking = bookingRepository.save(booking);
```

Comment:
you work with object, don't need to create new variable
```suggestion
        bookingRepository.save(booking);
```

Reply:
not fixed